s3/pkg/service/bucket.go

```diff
@@ -297,3 +298,40 @@ func (s *s3Service) ListBucketLifecycle(ctx context.Context, in *pb.BaseRequest,
 	log.Info("list lifecycle successfully")
 	return nil
 }
+
+func (s *s3Service) PutBucketACL(ctx context.Context, in *pb.PutBucketACLRequest, out *pb.BaseResponse) error {
+	log.Info("PutBucketACL is called in s3 service.")
+	var err error
+	defer func() {
+		out.ErrorCode = GetErrCode(err)
+	}()
+
+	bucket, err := s.MetaStorage.GetBucket(ctx, in.ACLConfig.BucketName, true)
+	if err != nil {
+		log.Errorf("failed to get bucket meta. err: %v\n", err)
+		return err
+	}
+
+	isAdmin, tenantId, err := util.GetCredentialFromCtx(ctx)
+	if err != nil && isAdmin == false {
+		log.Error("get tenant id failed")
+		err = ErrInternalError
+		return nil
+	}
+
+	if isAdmin == false {
+		if bucket.TenantId != tenantId {
+			err = ErrBucketAccessForbidden
+			return err
+		}
+	}
+
+	bucket.Acl = &pb.Acl{CannedAcl: in.ACLConfig.CannedAcl}
+	err = s.MetaStorage.Db.PutBucket(ctx, bucket)
+	if err != nil {
+		log.Error("failed to put bucket, err:", err)
+		return nil
+	}
+	log.Infoln("Put bucket acl successfully.")
+	return nil
```

Comment:
In this function, when error happened, some times returned err, sometimes returned nil, please check, thanks.

Reply:
@sfzeng , thanks for your review,  I will fix it to return error always when error happened